### PR TITLE
feat: #120 Create form component

### PR DESCRIPTION
**Pull request by Lelik7777**

### CODE REVIEW

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/form/formField/FormField.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import {ReactElement} from "react";
+import {Field as FieldForm} from "@radix-ui/react-form";
+import {FieldContainer} from "src/component/form/formField/fieldContainer/FieldContainer";
+import {FormControl} from "src/component/form/formField/formControl/FormControl";
+import styles from "src/component/form/formField/FormField.module.scss";
+
+interface FormFieldProps{
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/component/form/formField/FormField.tsx:7]: Add rule to eslint/prettier, not just add space pls

File: mw-webapp/src/component/form/formField/fieldContainer/FieldContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+/**
+ * Component contains label,messages for input data
+ */
+export const FieldContainer:React.FC<FieldContainerProps> = (props:FieldContainerProps) => {
+
+  return(
+    <div className={styles.container}>
+      <FormLabel>
```
Comment (Shustovsky) [inline mw-webapp/src/component/form/formField/fieldContainer/FieldContainer.tsx:27]: it seems to me that this component is completely superfluous here, and you can use {Label as LabelForm} directly here

File: mw-webapp/src/component/form/formField/fieldContainer/formMessage/FormMessage.module.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.message {
+  font-size: 13px;
+  color: #000;
+  opacity: 0.8;
+}
```
Comment (Shustovsky) [inline mw-webapp/src/component/form/formField/fieldContainer/formMessage/FormMessage.module.scss:5]: please use variables

File: mw-webapp/src/component/form/formField/fieldContainer/formLabel/FormLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import {PropsWithChildren} from "react";
+import {Label as LabelForm} from "@radix-ui/react-form";
+import styles from "src/component/form/formField/formLabel/FormLabel.module.scss";
+
+
+interface FormLabelProps{
+  /**
```
Comment (Shustovsky) [inline mw-webapp/src/component/form/formField/fieldContainer/formLabel/FormLabel.tsx:7]: please use Eslint

**Comment by Shustovsky** [comment]
can I find out why you are doing PR from your personal repository?

File: mw-webapp/src/router/Router.tsx [hunk 5]
```diff
@@ -12,6 +12,10 @@ export const Router = () => {
         path={pages.auth.path}
         element={pages.auth.element}
       />
+      <Route
+        path={pages.waysPage.path}
+        element={pages.waysPage.element}
+      />
       <Route
         path={pages.users.path}
         element={pages.users.element}
```
Comment (Shustovsky) [inline mw-webapp/src/router/Router.tsx:21]: these files are not related to Create form component

File: mw-webapp/src/router/pages.tsx [hunk 6]
```diff
@@ -15,6 +16,10 @@ export const pages = {
     path: "main",
     element: <MainPage />,
   },
+  waysPage: {
+    path: "waysPage",
+    element: <WaysPage />,
+  },
```
Comment (Shustovsky) [inline mw-webapp/src/router/pages.tsx:22]: these files are not related to Create form component

File: mw-webapp/src/component/form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      <FormField
+        name={"email"}
+        labelContent="email"
+        messageValueMissing="Please enter your email"
+        messageTypeMismatch="Please provide a valid email"
+        controlContent={
+          <input
+            type='email'
+            required
+          />}
```
Comment (SergioAJS) [inline mw-webapp/src/component/form/Form.tsx:30]: Maybe it's better to use Input component from our components?
You can see it if you update your working branch with the latest changes from main branch

**Pull request closed without merging**